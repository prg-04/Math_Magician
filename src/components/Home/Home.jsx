import React from 'react';
import { Link } from 'react-router-dom';

const Home = () => (
  <div className="home__hero">
    <div className="home__hero-intro">
      <h2>Welcome to Math Magician!</h2>
      <p>
        Math Magician is a website that offers a variety of mathematical tools
        and resources. Whether you need to perform basic calculations or find
        inspiration through insightful quotes, Math Magician has got you
        covered.
      </p>
    </div>

    <div className="home__hero-calc">
      <h2>Calculator</h2>
      <p>
        Visit the Calculator page to perform basic arithmetic calculations
        quickly and easily. It provides a user-friendly interface for addition,
        subtraction, multiplication, and division operations.
      </p>
      <Link to="/calculator" className="cta-button">
        Go to Calculator
      </Link>
    </div>

    <div className="home__hero-quote">
      <h2>Quote of the Day</h2>
      <p>
        Discover a new quote every day to inspire your mathematical journey. Our
        curated collection of quotes from mathematicians, scientists, and
        philosophers will keep you motivated.
      </p>
      <Link to="/quotes" className="cta-button">
        See Quote of the Day
      </Link>
    </div>
  </div>
);

export default Home;
