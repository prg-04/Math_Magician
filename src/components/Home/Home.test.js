import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Home from './Home';
import '@testing-library/jest-dom/extend-expect';
// Import the toBeInTheDocument matcher
describe('Home', () => {
  test('renders the welcome message', () => {
    render(<Home />, { wrapper: MemoryRouter });
    const welcomeMessage = screen.getByText(/Welcome to Math Magician/i);
    expect(welcomeMessage).toBeInTheDocument();
  });
  test('renders the calculator section', () => {
    render(<Home />, { wrapper: MemoryRouter });
    const calculatorSection = screen.getByRole('heading', {
      name: /Calculator/i,
    });
    expect(calculatorSection).toBeInTheDocument();
    const goToCalculatorLink = screen.getByRole('link', {
      name: /Go to Calculator/i,
    });
    expect(goToCalculatorLink).toBeInTheDocument();
    expect(goToCalculatorLink).toHaveAttribute('href', '/calculator');
  });
  test('renders the quote section', () => {
    render(<Home />, { wrapper: MemoryRouter });
    const quoteSection = screen.getByRole('heading', {
      name: /Quote of the Day/i,
    });
    expect(quoteSection).toBeInTheDocument();
    const seeQuoteLink = screen.getByRole('link', {
      name: /See Quote of the Day/i,
    });
    expect(seeQuoteLink).toBeInTheDocument();
    expect(seeQuoteLink).toHaveAttribute('href', '/quotes');
  });
});
