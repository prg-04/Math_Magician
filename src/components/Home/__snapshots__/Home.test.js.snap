// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<DocumentFragment>
  <div
    class="home__hero"
  >
    <div
      class="home__hero-intro"
    >
      <h2>
        Welcome to Math Magician!
      </h2>
      <p>
        Math Magician is a website that offers a variety of mathematical tools and resources. Whether you need to perform basic calculations or find inspiration through insightful quotes, Math Magician has got you covered.
      </p>
    </div>
    <div
      class="home__hero-calc"
    >
      <h2>
        Calculator
      </h2>
      <p>
        Visit the Calculator page to perform basic arithmetic calculations quickly and easily. It provides a user-friendly interface for addition, subtraction, multiplication, and division operations.
      </p>
      <a
        class="cta-button"
        href="/calculator"
      >
        Go to Calculator
      </a>
    </div>
    <div
      class="home__hero-quote"
    >
      <h2>
        Quote of the Day
      </h2>
      <p>
        Discover a new quote every day to inspire your mathematical journey. Our curated collection of quotes from mathematicians, scientists, and philosophers will keep you motivated.
      </p>
      <a
        class="cta-button"
        href="/quotes"
      >
        See Quote of the Day
      </a>
    </div>
  </div>
</DocumentFragment>
`;
